pandas/core/arrays/datetimelike.py

```diff
@@ -2103,7 +2103,7 @@ def _round(self, freq, mode, ambiguous, nonexistent):
         values = self.view("i8")
         values = cast(np.ndarray, values)
         nanos = to_offset(freq).nanos  # raises on non-fixed frequencies
-        nanos = delta_to_nanoseconds(to_offset(freq), self._reso)
+        nanos = delta_to_nanoseconds(to_offset(freq), self._creso)
```

Comment:
This discards the result of the previous line. I wonder why the previous line is still needed?

Reply:
the previous line likely shouldn't assign `nanos =`.  It is just accessing `.nanos` so that we get the right exception message if that raises

Reply:
> ... we get the right exception message if that raises

That's exactly how I got there.